cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelper.java

```diff
@@ -151,8 +151,14 @@ private void setFollowerThrottledRateIfUnset(int brokerId) {
   }
 
   private void setThrottledRateIfUnset(int brokerId, String configKey) {
-    assert (_throttleRate != null);
-    assert (configKey.equals(LEADER_THROTTLED_RATE) || configKey.equals(FOLLOWER_THROTTLED_RATE));
+    if (_throttleRate == null) {
+      throw new IllegalStateException("Throttle rate cannot be null");
+    }
+    if ((!configKey.equals(LEADER_THROTTLED_RATE)) && (!configKey.equals(FOLLOWER_THROTTLED_RATE))) {
```

Comment:
1. Superfluous parentheses around (1) `!configKey.equals(LEADER_THROTTLED_RATE)` and (2) `!configKey.equals(FOLLOWER_THROTTLED_RATE)`. Can we drop them for better readability?

* Please see https://docs.oracle.com/javase/tutorial/java/nutsandbolts/operators.html

2. Nit: This is a private function of the class -- i.e. no public / protected / package private access. Hence, I feel that this sanity check is redundant all together, but the decision to keep or remove it is up to you.

Reply:
1. I will remove the superfluous parentheses.

2. I will keep this check. I don't think of it as a sanity check since it's not checking input parameters. Instead, it's basically saying "if at this point of the execution, the `_throttleRate ` field is null, that means we have an illegal state". Therefore, it has nothing to do with whether this method is public/protected/private.

Reply:
> I will keep this check. I don't think of it as a sanity check since it's not checking input parameters. Instead, it's basically saying "if at this point of the execution, the _throttleRate field is null, that means we have an illegal state". Therefore, it has nothing to do with whether this method is public/protected/private.

My comment is about `configKey`, which is an input to this private function, -- i.e. not about `_throttleRate`.

Reply:
I see. OK, I will remove this sanity check.